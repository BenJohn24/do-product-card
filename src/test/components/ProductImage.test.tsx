import { render, screen } from "@testing-library/react"
import ProductCard, { ProductImage } from "../../components"
import { product2 } from "../data/products"

describe('ProductImage', () => {
    test('debe mostrar el component con la imagen correspondiente ', () => {
        const { img} = product2
      render(<ProductImage img={ img } />)

      const imgElement = screen.getByRole('img');

      expect(imgElement).toHaveAttribute('src', img);
    });

    test('debe de mostrar el componente con una imagen', () => {
      const { asFragment } = render( <ProductCard product={ product2 }>
     
        {
            ()=> (

                <ProductImage />
            )
        }
     
      </ProductCard>
      );

      expect(asFragment()).toMatchSnapshot();
    })
    
    
})