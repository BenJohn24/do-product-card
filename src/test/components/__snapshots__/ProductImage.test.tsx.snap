// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductImage > debe de mostrar el componente con una imagen 1`] = `
<DocumentFragment>
  <div
    class="_productCard_133f98 undefined"
  >
    <img
      alt="Product Image"
      class="_productImg_133f98 undefined "
      src="./coffee-mug.png"
    />
  </div>
</DocumentFragment>
`;
